### PR TITLE
chore: update ReadTheDocs config file to be compliant

### PR DESCRIPTION
**Pull request by matthewfeickert**
@alexander-held not sure if this is going to fix your issues, but this is basically the [`pyhf` RTD config](https://github.com/scikit-hep/pyhf/blob/4ec7bf357323d1c8084c63db41bbaa40d599a04d/.readthedocs.yaml) that has been working fine so far.

* c.f. https://twitter.com/readthedocs/status/1522266722635436041
* https://github.com/scikit-hep/pyhf/pull/2267

### CODE REVIEW

**Comment by matthewfeickert** [comment]
Also renamed as

> [Read the Docs supports configuring your documentation builds with a configuration file. This file is named .readthedocs.yaml and should be placed in the top level of your Git repository.](https://docs.readthedocs.io/en/stable/config-file/v2.html)

**Comment by matthewfeickert** [comment]
@alexander-held There's no RTD test build happening at the moment. Do you have it configured to not run on PRs?

File: .readthedocs.yaml [hunk 1]
```diff
@@ -5,17 +5,25 @@
...
 version: 2
 
+# Set the version of Python and other tools you might need
+build:
+  os: ubuntu-22.04
+  tools:
+    python: "3.11"
+  apt_packages:
+    - curl
+    - jq
```
Comment (alexander-held) [inline .readthedocs.yaml:15]: I don't think either of these are required (`pyhf` uses them for something probably?).

**Comment by alexander-held** [comment]
> @alexander-held There's no RTD test build happening at the moment. Do you have it configured to not run on PRs?

Yes, so far it was just running on `master` as there was never a need for branches (local compilation has been working fine for testing). I now switched the setup temporarily to build from this branch, but still see the build failing.

These are the commands I see running:
```bash
git clone --depth 1 https://github.com/scikit-hep/cabinetry.git .
git fetch origin --force --prune --prune-tags --depth 50 refs/heads/chore/update-RTD-config:refs/remotes/origin/chore/update-RTD-config
git checkout --force origin/chore/update-RTD-config
git clean -d -f -f
cat .readthedocs.yaml
apt-get update --assume-yes --quiet
apt-get install --assume-yes --quiet -- curl jq
asdf global python 3.11.4
python -mvirtualenv $READTHEDOCS_VIRTUALENV_PATH
python -m pip install --upgrade --no-cache-dir pip setuptools
python -m pip install --upgrade --no-cache-dir pillow mock==1.0.1 alabaster>=0.7,<0.8,!=0.7.5 commonmark==0.9.1 recommonmark==0.5.0 sphinx<2 sphinx-rtd-theme<0.5 readthedocs-sphinx-ext<2.3 jinja2<3.1.0
python -m pip install --upgrade --upgrade-strategy only-if-needed --no-cache-dir .[docs,contrib]
cat docs/conf.py
python -m sphinx -T -E -b html -d _build/doctrees -D language=en . $READTHEDOCS_OUTPUT/html
```

The thing I do not understand here is what triggers this line
```
python -m pip install --upgrade --no-cache-dir pillow mock==1.0.1 alabaster>=0.7,<0.8,!=0.7.5 commonmark==0.9.1 recommonmark==0.5.0 sphinx<2 sphinx-rtd-theme<0.5 readthedocs-sphinx-ext<2.3 jinja2<3.1.0
```

This installs `sphinx-rtd-theme<0.5` which does not subsequently get updated, as the dependency listed in `cabinetry` has no lower bound on this:
https://github.com/scikit-hep/cabinetry/blob/2af877c58052c335572fcea4e604d49f32804e0a/setup.py#L27-L31

I imagine this could be solved by adding a lower bound, I'll push a commit to this branch here to try it out.

It is very unclear to me why this one step happens with the upper bounds set though (which is what ultimately causes the problem). Maybe @henryiii might have an idea?

File: .readthedocs.yaml (hunk 1 above)
Comment (matthewfeickert) [reply .readthedocs.yaml]: Yeah, I just left them in. Removing now.

**Comment by matthewfeickert** [comment]
I agree that's strange, but I also agree that this is probably not worth the effort to debug.

**Comment by henryiii** [comment]
RtD pre-installs old versions of things, like the old rtd theme. Setting a minimum forces the upgrade.

**Review by alexander-held — APPROVED** [review]
Thanks everyone, this looks good to me then.

```
* add build.os config setting for readthedocs
* set lower bound of readthedocs Sphinx theme to fix readthedocs CI
```

**Pull request merged**